actions/inventory.go

```diff
@@ -341,14 +341,11 @@ func (a *InventoryCollectorAction) setDefaultAttributes() {
 
 // CollectDrives executes drive collectors and merges the data into device.[]*Drive
 // nolint:gocyclo // TODO(joel) if theres more conditionals to be added in here, the method is to be split up.
-func (a *InventoryCollectorAction) CollectDrives(ctx context.Context) error {
-	// nolint:errcheck // deferred method catches a panic, error check not required.
-	defer func() error {
+func (a *InventoryCollectorAction) CollectDrives(ctx context.Context) (err error) {
```

Comment:
I wanted to make sure you saw these @joelrebel. These are the only changes that potentially impact runtime. This is a change in behavior because now `a.failOnError` is actually going to influence the returned error. My first reaction to realizing that this code was wrong was to just do `defer func() { _ = recover() }()` but then thought maybe I should do what was intended instead. Would love your thoughts here.

Reply:
When you first pointed this out, I thought the initial code was incorrectly overwritting the returned error. 
But it seems like the `return nil` in the initial code defer func would not overwrite the returned error.

I tried both variants and they seem to behave the same https://go.dev/play/p/ktvkitbWyh5
so the change looks fine to me, let me know if Im missing something obvious.

Reply:
Yep it seems to me that returns from defer'd functions are just silently dropped